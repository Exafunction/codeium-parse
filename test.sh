#!/bin/bash
set -euo pipefail

cd "$(dirname "${BASH_SOURCE[0]}")"
for test_file in test_files/*; do
    test_file="$(basename "$test_file")"
    echo "$test_file"
    ./parse -file "test_files/$test_file" -use_tags_query -tags_query_dir "queries" > "goldens/$test_file.golden.tmp"
    trap 'rm "goldens/$test_file.golden.tmp"' EXIT
    diff -u "goldens/$test_file.golden" "goldens/$test_file.golden.tmp"
done
