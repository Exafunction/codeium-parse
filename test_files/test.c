#include "runtime/parser.h"

typedef struct {
  Parser* parser;
  TSSymbol lookahead_symbol;
} SkipPrecedingTreesSession;

static void parser__push(Parser* self, StackVersion version, Tree* tree,
                         TSStateId state) {
  ts_stack_push(self->stack, version, tree, false, state);
  ts_tree_release(tree);
}

static Tree* parser__lex(Parser* self, StackVersion version) {
  TSStateId parse_state = ts_stack_top_state(self->stack, version);
  Length start_position = ts_stack_top_position(self->stack, version);
  TSLexMode lex_mode = self->language->lex_modes[parse_state];
}
