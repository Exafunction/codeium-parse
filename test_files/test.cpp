#include <string>

namespace bar {
// Declaration.
bool baz(const std::string& bing);

struct Foo2;

}  // namespace bar

namespace bar::bar2 {
struct Foo3;
}

class Foo {
 public:
  // foo
  // foo2
  void foo() {
    // Do nothing.
  }
};

struct bar::Foo2 {};
struct bar::bar2::Foo3 {};

// baz
bool bar::baz(const std::string& bing) {
  // This shouldn't be recognized as a function.
  Foo dist();
  return bing.empty();
}

// one_pointer comment
bool* one_pointer() { return nullptr; }
// two_pointer comment
bool** two_pointer() { return nullptr; }

// Comment.
template <typename T>
class TemplatedFoo {
  template <typename U>
  void foo();
};

// Comment.
template <typename T>
template <typename U>
void TemplatedFoo<T>::foo() {}

// Comment.
template <>
template <>
void TemplatedFoo<int>::foo<int>() {}

// Comment.
template <typename T>
class TemplatedFoo2;

// Comment.
template <>
class TemplatedFoo2<int> {};
