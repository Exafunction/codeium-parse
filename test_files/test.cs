// clang-format off
namespace Namespace {
  // Comment A.
  internal delegate void A(params int[] test);
  // Comment F.
  public struct F<T>
      where T : struct {
  }
  // Comment F.
  [Nice]
  private record F<T1, T2>
      where T1 : I1, I2, new()
      where T2 : I2 {}
  // Comment Teacher.
  record Teacher(string FirstName, string LastName, string Subject) : Person(FirstName, LastName);
  // Comment B.
  enum B { Ten = 10, Twenty = 20 }
  // Comment F.
  public class F : object, IAlpha, IOmega {}
  // Comment Class.
  public partial class Class<in TParam>
      where TParam : class ?, notnull, F ? {
    // Comment operator +.
    public static int operator +(A a) {
      return 0;
    }
    // Comment GetSet.
    uint GetSet { get; set; }
    // Comment Foo.
    static extern Foo() => bar = 0;
    // Comment ~Class.
    extern ~Class() {}
    // Comment Bar
    public void Bar() => bar = 0;
  }
  // Comment IFoo.
  public interface IFoo {}
}
// clang-format on
