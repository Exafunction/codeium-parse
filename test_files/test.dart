import 'package:test/test.dart';

// Functions

// foo not a docstring
void foo() {
    print("Hello, World!");
}

/// bar is a docstring
int bar() {
    return 0;
}

// Classes with methods

class FooClass {
    int someField = 0;

    // Constructor
    FooClass() {
        this.someField = 0;
    }

    /// FCB docstring
    int fooClassBar() {
        return 0;
    }
}

// Mixins
mixin FooMixin {
    int barMixin() {
        return 0;
    }
}

// Combinations
class FooBar extends FooClass with Foo {
    int bar() {
        return 0;
    }
}

void main() {
    print("Hello, World!");
}
