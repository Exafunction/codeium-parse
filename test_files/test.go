package test_files

// Pattern 0.
type Pattern0 struct {
	a string
}

// Pattern 1.
type Pattern1 interface {
	a() string
}

// Unsupported.
type (
	UnsupportedStruct    struct{}
	UnsupportedInterface interface{}
)

func Pattern2() {}

// Comments
// here.
func (p *Pattern0) Method(string) {}

func (p Pattern0) Method2(a string) string {
	Pattern2()
	s := &Pattern0{}
	s.Method("")
	((*Pattern0)(s)).Method("")
	var s2 any
	s2.(*Pattern0).Method("")
	return a
}

type Alias string
