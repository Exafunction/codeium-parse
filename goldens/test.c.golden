Name: "runtime/parser.h"
Definition (definition.import):
#include "runtime/parser.h"



Name: parser__push
Parameters: (Parser* self, StackVersion version, Tree* tree,
                         TSStateId state)
Return type: void
Definition (definition.function):
static void parser__push(Parser* self, StackVersion version, Tree* tree,
                         TSStateId state) {
  ts_stack_push(self->stack, version, tree, false, state);
  ts_tree_release(tree);
}

Name: parser__lex
Parameters: (Parser* self, StackVersion version)
Return type: Tree
Definition (definition.function):
static Tree* parser__lex(Parser* self, StackVersion version) {
  TSStateId parse_state = ts_stack_top_state(self->stack, version);
  Length start_position = ts_stack_top_position(self->stack, version);
  TSLexMode lex_mode = self->language->lex_modes[parse_state];
}
